src/parser/typed.fs

```diff
@@ -17,63 +17,61 @@ and QType =
     | QInt
 
 type C =
```

Comment:
Pushed types to be specified, instead at the individual expression, at the union of Classic and Quantum expression. This makes it a lot easier to extract them. I think it still works since `typecheck` will return the union, so it always includes the type.